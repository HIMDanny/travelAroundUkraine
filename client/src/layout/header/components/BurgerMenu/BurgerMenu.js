import React, { useState } from 'react';
import { Box, MenuItem, Menu, IconButton } from '@mui/material';
import MenuIcon from '@mui/icons-material/Menu';
import { useSelector, useDispatch } from 'react-redux';
import { NavLink, useNavigate } from 'react-router-dom';
import { setIsModalOpen } from '../../../../store/slices/userSlice/userSlice';

const loggedPages = [
  {
    name: 'home',
    link: '/',
    innerContent: 'Home',
  },
  {
    name: 'catalogue',
    link: '/catalogue',
    innerContent: 'Catalogue',
  },
  {
    name: 'account',
    link: '/user',
    innerContent: 'Account',
  },
  {
    name: 'cart',
    link: '/cart',
    innerContent: 'Cart',
  },
  {
    name: 'logOut',
    link: null,
    innerContent: 'LogOut',
    onClick: (dispatch) => dispatch(setIsModalOpen(true)),
  },
];

const unloggedPages = [
  {
    name: 'home',
    link: '/',
    innerContent: 'Home',
  },
  {
    name: 'catalogue',
    link: '/catalogue',
    innerContent: 'Catalogue',
  },
  {
    name: 'logIn',
    link: '/login',
    innerContent: 'LogIn',
  },
  {
    name: 'cart',
    link: '/cart',
    innerContent: 'Cart',
  },
  {
    name: 'favorites',
    link: '/favorites',
    innerContent: 'Favorites',
  },
];

const boxSX = {
  flexGrow: 1,
  display: { xs: 'flex', laptop: 'none' },
  justifyContent: 'flex-end',
};

const menuSX = {
  display: { xs: 'flex', laptop: 'none' },
};

const menuItemSX = {
  padding: '6px 16px',
  width: '40vw',
  maxWidth: '200px',
  height: '50px',
  color: 'text.primary',
  textDecoration: 'none',
};

const anchorOrigin = {
  vertical: 'bottom',
  horizontal: 'left',
};

const transformOrigin = {
  vertical: 'top',
  horizontal: 'left',
};

const BurgerMenu = () => {
  const [anchorElNav, setAnchorElNav] = useState(null);
  const { isLogin } = useSelector((store) => store.userReducer);

  const navigate = useNavigate();
  const dispatch = useDispatch();

  const currentPages = isLogin ? loggedPages : unloggedPages;

  const handleOpenNavMenu = (event) => {
    setAnchorElNav(event.currentTarget);
  };

  const handleCloseNavMenu = () => {
    setAnchorElNav(null);
  };

  return (
    <Box sx={boxSX}>
      <IconButton
        size="large"
        aria-label="appbar"
        aria-controls="menu-appbar"
        aria-haspopup="true"
        onClick={handleOpenNavMenu}
        sx={{ padding: 0, color: 'text.primary' }}
      >
        <MenuIcon />
      </IconButton>

      <Menu
        id="menu-appbar"
        anchorEl={anchorElNav}
        anchorOrigin={anchorOrigin}
        keepMounted
        transformOrigin={transformOrigin}
        open={!!anchorElNav}
        onClose={handleCloseNavMenu}
        sx={menuSX}
      >
        {currentPages.map(({ name, link, innerContent, onClick }) => (
          <MenuItem
            component={NavLink}
            key={name}
            onClick={() => {
              if (onClick) {
                onClick(dispatch, navigate);
              }
              handleCloseNavMenu();
            }}
            to={link}
            sx={menuItemSX}
            end
          >
            {innerContent}
          </MenuItem>
        ))}
      </Menu>
    </Box>
  );
};

export default BurgerMenu;
