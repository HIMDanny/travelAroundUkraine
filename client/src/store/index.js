import { configureStore } from '@reduxjs/toolkit';
import userReducer from './slices/userSlice/userSlice';
import tourReducer from './slices/tourSlice/tourSlice';
import catalogueReducer from './slices/catalogueSlice/catalogueSlice';
import cartReducer from './slices/cartSlice/cartSlice';
import filterReducer from './slices/filterSlice/filterSlice';
import orderSlice from './slices/orderSlice/orderSlice';
import searchBarReducer from './slices/searchBarSlice/searchBarSlice';

const store = configureStore({
  reducer: {
    user: userReducer,
    tour: tourReducer,
    catalogue: catalogueReducer,
    cart: cartReducer,
    filter: filterReducer,
    order: orderSlice,
    searchBar: searchBarReducer,
  },
});

export default store;
