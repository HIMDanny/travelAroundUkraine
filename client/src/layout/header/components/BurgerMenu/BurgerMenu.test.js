import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../../../../store';
import BurgerMenu from './BurgerMenu';

jest.mock('@mui/material/Box', () => ({ children }) => <div>{children}</div>);
jest.mock('@mui/material/MenuItem', () => ({ children }) => <li>{children}</li>);
jest.mock('@mui/material/Menu', () => ({ children }) => <ul>{children}</ul>);
jest.mock('@mui/material/IconButton', () => ({ children }) => <button>{children}</button>);
jest.mock('@mui/material/Typography', () => ({ children }) => <a>{children}</a>);

describe('Render BurgerMenu', () => {
  test('should BurgerMenu render', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <BrowserRouter>
          <BurgerMenu isLogin />
        </BrowserRouter>
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
