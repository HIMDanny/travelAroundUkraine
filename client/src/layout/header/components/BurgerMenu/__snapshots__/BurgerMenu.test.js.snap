// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render BurgerMenu should BurgerMenu render 1`] = `
<DocumentFragment>
  <div>
    <button>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="MenuIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </button>
    <ul>
      <li>
        Home
      </li>
      <li>
        Catalogue
      </li>
      <li>
        LogIn
      </li>
      <li>
        Cart
      </li>
      <li>
        Favorites
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
